sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -328,6 +328,8 @@ object PullupCorrelatedPredicates extends Rule[LogicalPlan] with PredicateHelper
     // Only a few unary nodes (Project/Filter/Aggregate) can contain subqueries.
     case q: UnaryNode =>
```

Comment:
shall we make `Filter`, `Aggregate` and `Project` extend `SupportsSubquery` and only match `SupportsSubquery` here?

Reply:
Sounds good to me. We can also simplify the check inside `CheckAnalysis` in a follow-up PR as well.

Let me submit a separate PR for this one.

Reply:
Actually, we can get this one in first. How does it sound, @cloud-fan?